### ISSUE TITLE
🚀 Open API v3 is in General Release!

### ISSUE THREAD

**Issue by capndesign** [etsy/open-api#149]
### We’re ready for everybody to start exploring Open API v3! 

We’ve reached a major milestone — the vast majority of our endpoints are now in the general release phase. This means we feel confident that these endpoints are ready for use in a production environment. We can’t wait to see what you build with it.

### See what’s new

You can now explore [the new endpoints](https://developers.etsy.com/documentation/reference) and [documentation](https://developers.etsy.com/documentation). Many of the endpoints in the new version are the same as in v2, and we’ve updated our documentation to make it easier to understand.

Check out our [summary of changes](https://developers.etsy.com/documentation/migration/summary-changes) to see what’s new. If you discover an issue, just [submit a ticket](https://github.com/etsy/open-api/issues/new/choose).

### Migrate your app

The v2 version of the API will be available through at least the end of 2021, but we recommend migrating to v3 as soon as you can.This ensures your app uses the latest and greatest version of our API. Migrating early also helps bring your feedback to the top of the queue. We’ll let you know when we plan to deprecate v2.

### Thank you and what’s next 

We’ve said it in previous messages, but thank you to everyone who has submitted their feedback. It’s been critical to our understanding of what’s important to our developer community.

While the general release is an important milestone, we recognize it may not offer all the features you need to support your customers. We encourage you to [keep submitting bugs and feature requests](https://github.com/etsy/open-api/issues/new/choose). We’ll do our best to incorporate your feedback.

**Comment by chris-rogala** [comment]
@capndesign I started to migrate our systems to v3, and I noticed some things that could prevent us from moving forward. With the holiday season right around the corner, I believe it is very unlikely these issues will be addressed in time for us to migrate and test by the end of 2021.

We will not have time to finish the migration once holiday orders start rolling in this October. I do hope that Etsy will consider changing their support window for v2 to July 2022. I have posted multiple issues over the last few weeks, but I have not seen a lot of responses so I am doubting these issues will be addressed before the end of 2021 and in time to allow us to consume the endpoints. We will not have time to test our endpoints to ensure our systems can handle the surge in orders this fall.

Thanks in advance for considering this change.

**Comment by turboturtleD** [comment]
> @capndesign I started to migrate our systems to v3, and I noticed some things that could prevent us from moving forward. With the holiday season right around the corner, I believe it is very unlikely these issues will be addressed in time for us to migrate and test by the end of 2021.
> 
> We will not have time to finish the migration once holiday orders start rolling in this October. I do hope that Etsy will consider changing their support window for v2 to July 2022. I have posted multiple issues over the last few weeks, but I have not seen a lot of responses so I am doubting these issues will be addressed before the end of 2021 and in time to allow us to consume the endpoints. We will not have time to test our endpoints to ensure our systems can handle the surge in orders this fall.
> 
> Thanks in advance for considering this change.

I have to agree here. Though it now sounds like v2 will at least be supported though the end of 2021, we are still only in the beta testing phase with v3.  Realistically we won’t be able to even consider beginning full migration until Q1 which would naturally lend itself to an end of Q2 sunset for v2 at the absolute soonest.

**Comment by kbasali** [comment]
How or where can we make suggestions to you to add more features to the new API version? I have several questions about why you are not adding some features to API and I am sure Etsy has its own reasons not to add them. So can we ask and can your team answer our questions?

**Comment by Gareth064** [comment]
@etsyachristensen whilst on the subject of the Google Group. Who does the authorization of posts and replies? And why do replies need to be authorized?

**Comment by jasonpincin** [comment]
@Gareth064 wrote:
> I personally think we should scrap the email address as an official way to contact Etsy (except for escalations maybe? quoting an issue number from here) about the API and continue with raising issues here. Also keep the Google community as it will pivot completely to people talking about v3 as time goes on.

@damianrclem wrote:
> just a heads up... when I emailed developer@etsy.com about the discontinued endpoints used in our application, they directed me to create an issue here on the GitHub repo. Maybe this has changed since my request was posted a couple of months ago. There are several postings about a v3 endpoints on the Google group and replied those should be directed here.

Hello; nice to meet everyone here - I'm very excited to work with our developer community. Within Etsy, the Open API has recently become part of the broader API Platform group, which is focused on advancing all of our APIs (internal / public) and improving the developer experience of using them. As part of this, we're re-evaluating existing milestones and ways of working. You'll see more formal communications with detailed information soon. 

Related to the comments above, there may be some ambiguity around the methods of interaction which we'll work to clear up. I'm very committed to keeping this Etsy Open API Github presence as the preferred manner of reporting issues, sharing information, and collaborating on solutions beyond the transition to Open API v3 and indefinitely.

**Comment by chris-rogala** [comment]
@jasonpincin I absolutely agree with you on your comment about continuing to use the GitHub repo to track bugs and features. The idea of simply closing it out seems to very similar to the previous iteration of the API. There was little to no transparency with Etsy and the development community. From my work consulting and our own private repos, GitHub is the ideal location for this. This will ensure Etsy continues to receive enhancement request and it would be a great location for Etsy to gain very valuable insights on ideas the community would love added.

**Comment by jasonpincin** [comment]
👋  Hello! We continue to appreciate the feedback and issue reports we are receiving from the community and we're working through a review of each. For the remainder of this year we do not intend to make significant changes; however, you can expect the pace of issue resolution & communication to pick back up in January. 

If you haven't already, please take note of some recent adjustments to the [v2 sunset timeline](https://developers.etsy.com/documentation/migration/index/#launch-stages).

**Comment by Gareth064** [comment]
@jasonpincin How do we developers know when you have released a new version of or made changes to the API?

I see all over these issues some of your devs saying things like "we have released a fix for this today" with no other information about it.

I am building a .NET Wrapper around the API, which I started months ago and it just dawned on me that things may have changed on endpoints since I built my .NET code for them. I used a JSON Compare Online App to compare a version of the OAS file I had downloaded months ago to the version that is there and now and it is suggesting there are 162 differences. 😢

It would be very useful to have some sort of Release\Change Notes Issue pinned to allow you guys to let us know when you are changing or adding things.

Happy Holidays 🎄

**Comment by nateswart** [comment]
Hello everyone 👋 

I wanted to take a moment to introduce myself as well. Etsy recently hired me to join as Product Manager for the API Platforms and will be working alongside @jasonpincin on projects including Etsy's Open API. We greatly appreciate your feedback. It echos some of the conversations we've been having internally as well, which is a good confirmation we're discussing the right topics on how to improve developer experiences.

Specifically, to @Gareth064's point I'm working through how we can improve comms and awareness.
I'm just getting up to speed, but as @jasonpincin mentioned, the new year will be bringing increased pace and communications here and I look forward to growing the platform with everyone!

**Comment by Gareth064** [comment]
Hi @nateswart and welcome onboard. The 2 things which I will find useful from a comms perspective are:

1. Release/Change Notes for every push to the Production Etsy Open API
2. A GitHub project, or something similar, to show that Issues raised here have been placed on a backlog to be worked on and preferably other status to show if they are being worked on etc.

I appreciate that the 2nd point above might be duplication of work for you and your team as you will be using you're own work tracking tools, but from our point of view, it really helps us plan our own work as well if we know a bug is in the pipeline to be fixed.

One final thing which is completely missing from this whole development process is the concept of a proper Development Shop to build our apps against instead of doing it against a live shop where we might not have enough data to test with etc. Not a comms thing I know but it would not only help us, but can also be a tool for you to release changes and updates to to allow us to test against as well before you release to Production.

**Comment by main24** [comment]
@kbasali I'm not a maintainer of Etsy :)

**Comment by kbasali** [comment]
> @kbasali I will ping our support department and have them forward to me if necessary.

Thank you so much, I got the answer but it is not what I want to hear :)

**Comment by kbasali** [comment]
Hello @etsyachristensen , Andrew
I've sent an email to developer@etsy.com
If you have time, can you check it, please? Email Title: Big Update

**Comment by etsyachristensen** [comment]
@kbasali I have requested that they forward the email to me.

**Comment by etsyachristensen** [comment]
@kbasali What email address would it be coming from? Our support team can't find the email. Can you try to re-send to developers@etsy.com?

**Comment by kbasali** [comment]
Hello @etsyachristensen Andrew

Yes, I sent it 2 times to this email address.
Now I sent it again and the Ticket was created. Ticket ID: #8556483
The Ticket ID was not created in the emails I sent before.

The title of the emails I sent before: Big Update
The title of the email I just sent: Please Forward this email to Andrew Christensen
Sorry again to bother you here.

**Comment by sdumper** [comment]
I see coupons was deprecated where can i find it's equivalent in api3? Also i'm struggling to locate the analytics api? Any help would be appreciated even to say it no longer exists.
